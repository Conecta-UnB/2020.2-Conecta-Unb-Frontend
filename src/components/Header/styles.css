.componentHeader div{
  display: flex;
  flex-direction: row;
}

.componentHeader  #pages{
  width: 100%;
  height: 75px;
  background: #018B44;
}
.componentHeader #ConectaUnb{
  color: #ffffff;
  font-size: 35px;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: auto;
  margin-right: auto;
}

.componentHeader a {
  color: #ffffff;
  font-weight: bold;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: auto;
  margin-right: auto;
  text-decoration: none;
}
.componentHeader a:hover{
  color: #0000009a;
  font-weight: bold;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: auto;
  margin-right: auto;
}

.componentHeader #img{
  background-color: #ffffff;
}

.componentHeader img{
  height: 35px;
  margin-left: auto;
}