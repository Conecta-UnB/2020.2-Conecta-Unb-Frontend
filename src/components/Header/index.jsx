import React, { useContext } from 'react';
import { Link } from 'react-router-dom';

import { NewsContext } from '../../contexts/news';

import './styles.css';
import unbFgaLogo from '../../assets/unbFgaLogo.jpeg';

export default function Header() {
  const { getAllNews } = useContext(NewsContext);
  return (
    <div className="componentHeader">
      <div id="pages">
        <div id="ConectaUnb">
          ConectaUnB
        </div>
        <Link to="/">
          Home
        </Link>
        <Link
          onClick={async () => {
            await getAllNews();
          }}
          to="/news"
        >
          Notícias
        </Link>
        <a href="none">
          Eventos
        </a>
        <a href="none">
          EJs
        </a>
        <a href="none">
          Fórum
        </a>
        <a href="none" className="headerPages">
          Perfil
        </a>
      </div>
      <div id="img">
        <img src={unbFgaLogo} alt="unbFgaLogoImage" />
      </div>
    </div>
  );
}
