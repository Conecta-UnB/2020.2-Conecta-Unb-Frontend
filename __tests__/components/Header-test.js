import React from 'react';
import { render } from '@testing-library/react';
import Header from '../../src/components/Header';

describe('testando loginforms', () => {
  test('deve ser igual a snapshot', () => {
    const { asFragment } = render(<Header />);
    expect(asFragment(<Header />)).toMatchSnapshot();
  });
});
