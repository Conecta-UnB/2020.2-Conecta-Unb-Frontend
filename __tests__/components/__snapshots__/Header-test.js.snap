// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testando loginforms deve ser igual a snapshot 1`] = `
<DocumentFragment>
  <div
    class="componentHeader"
  >
    <div
      id="pages"
    >
      <div
        id="ConectaUnb"
      >
        ConectaUnB
      </div>
      <a
        href="none"
      >
        Home
      </a>
      <a
        href="none"
      >
        Notícias
      </a>
      <a
        href="none"
      >
        Eventos
      </a>
      <a
        href="none"
      >
        EJs
      </a>
      <a
        href="none"
      >
        Fórum
      </a>
      <a
        class="headerPages"
        href="none"
      >
        Perfil
      </a>
    </div>
    <div
      id="img"
    >
      <img
        alt="unbFgaLogoImage"
        src="[object Object]"
      />
    </div>
  </div>
</DocumentFragment>
`;
